samcli/lib/remote_invoke/lambda_invoke_executors.py

```diff
@@ -154,3 +212,31 @@ def map(self, remote_invoke_input: RemoteInvokeExecutionInfo) -> RemoteInvokeExe
                 remote_invoke_input.response = boto_response.get(PAYLOAD)
 
         return remote_invoke_input
+
+
+class LambdaStreamResponseOutputFormatter(RemoteInvokeRequestResponseMapper):
+    """
+    This class helps to format streaming output response for lambda service that will be printed on the CLI.
+    It loops through EventStream elements and adds them to response, and once InvokeComplete is reached, it updates
+    log_output and response objects in remote_invoke_input.
+    """
+
+    def map(self, remote_invoke_input: RemoteInvokeExecutionInfo) -> RemoteInvokeExecutionInfo:
+        """
+        Maps the lambda response output to the type of output format specified as user input.
+        If output_format is original-boto-response, write the original boto API response
+        to stdout.
+        """
+        if remote_invoke_input.output_format == RemoteInvokeOutputFormat.DEFAULT:
+            LOG.debug("Formatting Lambda output response")
+            boto_response = cast(Dict, remote_invoke_input.response)
```

Comment:
We should start moving away from `Dict` and use `dict` instead. https://peps.python.org/pep-0585/ and https://docs.python.org/3/library/typing.html#typing.Dict

Not critical but will safe us in the longer run